src/EditColumnsModal/EditColumnsModal.tsx

```diff
@@ -0,0 +1,134 @@
+import { LfLocalizationService } from "@laserfiche/lf-js-utils";
+import React, { useEffect, useState } from "react";
+import { ColumnDef } from "@laserfiche/types-lf-ui-components";
+import "./EditColumnsModal.css";
+
+const resources: Map<string, object> = new Map<string, object>([
+  [
+    "en-US",
+    {
+      OK: "Ok",
+      CANCEL: "Cancel",
+      ADD_REMOVE_COLUMNS: "Add/Remove Columns",
+    },
+  ],
+  [
+    "es-MX",
+    {
+      OK: "Ok - Spanish",
+      CANCEL: "Cancel - Spanish",
+      ADD_REMOVE_COLUMNS: "Add/Remove Columns - Spanish",
+    },
+  ],
+]);
+
+const localizationService: LfLocalizationService = new LfLocalizationService(
+  resources
+);
+
+const OK = localizationService.getString("OK");
+const CANCEL = localizationService.getString("CANCEL");
+const ADD_REMOVE_COLUMNS: string =
+  localizationService.getString("ADD_REMOVE_COLUMNS");
+
+export default function EditColumnsModal(props: {
+  onClose: (columns?: ColumnDef[]) => void;
+  initialColumnsSelected: ColumnDef[];
+  allColumnOptions: ColumnDef[];
+}) {
+  const [open, setOpen] = useState<boolean>(false);
+  const [columnsSelected, setColumnsSelected] = useState<ColumnDef[]>(props.initialColumnsSelected);
+
+
+  const closeOnEscapeKeyDown = (keyboardEvent: KeyboardEvent) =>  {
+    if (keyboardEvent.code === "Escape") {
+      props.onClose();
+    }
+  };
+  useEffect(() => {
+    document.body.addEventListener("keydown", closeOnEscapeKeyDown);
+    setTimeout(() => {
+      setOpen(true);
+    });
+
+  }, []);
+
+  const closeDialog = (columns?: ColumnDef[]) => {
+    props.onClose(columns);
+  };
+
+  function onCheckboxChange(column: ColumnDef) {
+    // we provide a new array (not just new values) to trigger change detection
+    let newSelected: ColumnDef[];
+    if (columnsSelected.includes(column)) {
+      newSelected = [...columnsSelected.filter((c) => c != column)];
+    } else {
+      newSelected = [...columnsSelected];

```

Comment:
this is good, another way to do this would be
newSelected = newSelected.concat([column])